### DISCUSSION TITLE
Debug node ace - VSCode support

### DISCUSSION THREAD

**Discussion by Ninjeneer** [adonisjs/core#2884] [Help]
Hello,

I am trying to setup the VSCode debugger for an Adonis project. I am using the `node ace serve` command to run my project, and my breakpoints keep being ignored.

Here is my `.vscode/launch.json` configuration:
```json
{
  "type": "pwa-node",
  "request": "launch",
  "name": "Debug Adonis",
  "runtimeExecutable": "node",
  "runtimeArgs": [
    "ace",
    "serve",
    "--watch"
  ],
  "sourceMaps": true,
  "resolveSourceMapLocations": [
    "${workspaceFolder}/**",
    "!**/node_modules/**"
  ],
  "internalConsoleOptions": "openOnSessionStart"
}
```

**Node version** : v14.17.3

It appears that the `ace`file is debugged, but not the web server itself
Is there any trick here that I am missing ? What about adding a debug section in the documentation ?

Thanks !

**Comment by FelipeGangrel** [comment]
Try this

```json 
{
  "version": "0.2.0",
  "configurations": [
    {
      "type": "pwa-node",
      "request": "launch",
      "name": "Debug Application",
      "runtimeExecutable": "node",
      "runtimeArgs": ["ace", "serve", "--watch"],
      "internalConsoleOptions": "openOnSessionStart"
    }
  ]
}
```

**Reply by markgidman-rad** [reply]
Worked for me. Had to change `type` to `node` but worked great after that.

**Reply by PandiyanCool** [reply]
thanks guys!

**Reply by pomarec** [reply]
If you are trying to do this in a docker container, see : https://github.com/adonisjs/core/issues/4800#issuecomment-2491863629

**Reply by k1w1** [reply]
I am still seeing the problem in the original issue, despite following the directions [here](https://docs.adonisjs.com/guides/basics/debugging#vscode-debugger). I only see the source of ace.js, but not the rest of the application.

**Comment by Zenkylo** [comment]
![image](https://github.com/user-attachments/assets/00a3b126-3597-4199-9ea3-9a87fc3847f1)
You can also just run a _Javascript Debug Terminal_. This will open up a terminal where you can just run `npm run dev` like you would in any other terminal window.
